contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -994,34 +990,45 @@ sp_describe_undeclared_parameters_internal(PG_FUNCTION_ARGS)
 			list_length(pltsql_parse_result->body) != 2 ||
 			((PLtsql_stmt *) lsecond(pltsql_parse_result->body))->cmd_type != PLTSQL_STMT_EXECSQL)
 		{
-			ereport(ERROR,
-					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-					 errmsg("Unsupported use case in sp_describe_undeclared_parameters")));
+			is_supported_case_sp_describe_undeclared_parameters = false;
 		}
-		parsedbatch = ((PLtsql_stmt_execsql *) lsecond(pltsql_parse_result->body))->sqlstmt->query;
+		else
+		{
+			parsedbatch = ((PLtsql_stmt_execsql *) lsecond(pltsql_parse_result->body))->sqlstmt->query;
+
+			args = create_args(0);
+			if (params)
+				read_param_def(args, params);
 
-		args = create_args(0);
-		if (params)
-			read_param_def(args, params);
+			/* Next, pass the ANTLR-parsed batch to the backend parser */
+			sql_dialect_value_old = sql_dialect;
+			sql_dialect = SQL_DIALECT_TSQL;
+			raw_parsetree_list = pg_parse_query(parsedbatch);
+		}
 
-		/* Next, pass the ANTLR-parsed batch to the backend parser */
-		sql_dialect_value_old = sql_dialect;
-		sql_dialect = SQL_DIALECT_TSQL;
-		raw_parsetree_list = pg_parse_query(parsedbatch);
-		if (list_length(raw_parsetree_list) != 1)
+		if (is_supported_case_sp_describe_undeclared_parameters)
 		{
-			ereport(ERROR,
-					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-					 errmsg("Unsupported use case in sp_describe_undeclared_parameters")));
+			if (list_length(raw_parsetree_list) != 1)
+			{
+				is_supported_case_sp_describe_undeclared_parameters = false;
+			}
+			else
+			{
+				list_item = list_head(raw_parsetree_list);
+				parsetree = lfirst_node(RawStmt, list_item);
+			}
 		}
-		list_item = list_head(raw_parsetree_list);
-		parsetree = lfirst_node(RawStmt, list_item);
 
 		/*
 		 * Analyze the parsed statement to suggest types for undeclared
 		 * parameters
 		 */
-		switch (nodeTag(parsetree->stmt))
+		if (is_supported_case_sp_describe_undeclared_parameters)
```

Comment:
This can be combined with the above if statement?

Reply:
Updated